contrib/format-hdf5/src/main/java/org/apache/drill/exec/store/hdf5/HDF5Utils.java

```diff
@@ -0,0 +1,248 @@
+      // Convert HDF5 Enum to String
+      return new HDF5Attribute(MinorType.GENERIC_OBJECT, key, reader.enums().getAttr(pathName, key));
+    } else if (type.isAssignableFrom(BitSet.class)) {
+      return new HDF5Attribute(MinorType.BIT, key, reader.bool().getAttr(pathName, key));
+    }
+
+    logger.info("Reading attributes of type " + attributeInfo + " not yet implemented.");
+    return null;
+  }
+
+  public static MinorType getDataType(final HDF5DataSetInformation datasetInfo) {
+
+    HDF5DataTypeInformation typeInfo = datasetInfo.getTypeInformation();
+    Class<?> type = typeInfo.tryGetJavaType();
+    String name = typeInfo.getDataClass().name();
+    if (type == null) {
+      logger.info("Datasets of type " + typeInfo + " not implemented.");
+      //Fall back to string
+      if(name.equalsIgnoreCase("OTHER")) {
+        return MinorType.GENERIC_OBJECT;
+      } else {
+        return MinorType.VARCHAR;
+      }
+    } else if (type.isAssignableFrom(long.class)) {
+      return MinorType.BIGINT;
+    } else if (type.isAssignableFrom(short.class)) {
+      return MinorType.SMALLINT;
+    } else if (type.isAssignableFrom(byte.class)) {
+      return MinorType.TINYINT;
+    } else if (type.isAssignableFrom(int.class)) {
+      return MinorType.INT;
+    } else if (type.isAssignableFrom(double.class) || type.isAssignableFrom(float.class)) {
+      return MinorType.FLOAT8;
+    } else if (type.isAssignableFrom(String.class)) {
+      return MinorType.VARCHAR;
+    } else if (type.isAssignableFrom(java.util.Date.class)) {
+      return MinorType.TIMESTAMP;
+    } else if (type.isAssignableFrom(boolean.class) || type.isAssignableFrom(BitSet.class)) {
+      return MinorType.BIT;
+    } else if (type.isAssignableFrom(Map.class)) {
+      return MinorType.MAP;
+    } else if (type.isAssignableFrom(Enum.class)) {
+      return MinorType.GENERIC_OBJECT;
+    }
+    return MinorType.GENERIC_OBJECT;
+  }
+
+  protected static Class<?> getDatasetClass(String path, IHDF5Reader reader) {
+    HDF5DataSetInformation info = reader.getDataSetInformation(resolvePath(path, reader));
+    Class<?> dataClass = info.getTypeInformation().tryGetJavaType();
+    return dataClass;
+  }
+
+  protected static String resolvePath(String path, IHDF5Reader reader) {
+    if (reader.exists(path)) {
+      // Resolve references, if any.
+      if (reader.isDataSet(path)) {
+        HDF5DataClass dataClass = reader.getDataSetInformation(path).getTypeInformation().getDataClass();
+        if (dataClass.toString().equals("REFERENCE")) {
+          return reader.readObjectReference(path);
+        }
+      }
+      return path;
+    } else if (!caseSensitive) {
+      // Look for a match with a different case.
+      String[] pathParts = path.split("/");
+      String resolvedPath = "/";
+      for (int i = 1; i < pathParts.length; i++) {
+        String testPath = (resolvedPath.endsWith("/")) ? resolvedPath + pathParts[i] : resolvedPath + "/" + pathParts[i];
```

Comment:
This might cause problems on systems other than Linux since separator is different.

Reply:
Good point. Best to use the `File` or `Path` classes which are machine independent.

Reply:
So, this is actually a bit tricky.  This "path" isn't a file system path, but rather a path within the HDF5 file itself.  I'm testing on a Windows machine now, but in theory it shouldn't matter since this is within the HDF5 file and not an actual file system path.   If the tests are successful, I will resolve this comment, if not, I'll update the PR.

Reply:
I ran the tests on a Windows 10 VM and confirmed this.  The path is within HDF5 so it isn't an issue.